python/python_direct/runtime.cpp

```diff
@@ -0,0 +1,144 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <bindings.h>
+#include <python_utils.h>
+
+#include <fusion.h>
+#include <runtime/executor_kernel_arg.h>
+#include <runtime/fusion_executor_cache.h>
+#include <runtime/fusion_kernel_runtime.h>
+
+namespace nvfuser::python {
+
+// direct_bindings/runtime.cpp contains Fusion and FusionExecutorCache.
+// TODO IrContainer and Kernel would be implemented here.
+
+namespace {
+
+void bindFusion(py::module& nvfuser) {
+  py::class_<FusionGuard>(nvfuser, "FusionGuard")
+      .def(
+          py::init<Fusion*>(),
+          py::arg("fusion"),
+          R"(
+Create a new FusionGuard to manage the active fusion context.
+
+A FusionGuard is a RAII-style guard that sets the active fusion context for the current scope.
+When the guard is created, it sets the provided fusion as the active fusion.
+When the guard is destroyed, it restores the previous fusion context.
+
+Parameters
+----------
+fusion : Fusion
+    The fusion to set as the active fusion context.
+
+Examples
+--------
+>>> fusion = Fusion()
+>>> with FusionGuard(fusion):
+...     # Define fusion operations here
+...     t0 = ops.add(x, y)
+...     # The fusion context is automatically restored when exiting the with block
```

Comment:
the doc suggests context manager protocol, do we need to define __enter__ and __exit__ in order to use that?

Reply:
Removed. I don't think `FusionGuard` will be used in this way.